### PR TITLE
chore: Add kokoro nightly, presubmit and continuous job configurations

### PR DESCRIPTION
**Pull request by bajajneha27**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/ruby-spanner/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [ ] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [ ] Update code documentation if necessary.

closes: #<issue_number_goes_here>

### CODE REVIEW

File: .kokoro/continuous/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
+FROM ruby:3.1.1 AS build-env
```
Comment (dazuma) [inline .kokoro/continuous/Dockerfile:1]: Might as well use `ruby:3.1.2` or even `ruby:3.1`.

But more to the point, I'm not clear what this dockerfile is for...?

**Review by dazuma — DISMISSED** [review]

File: .kokoro/continuous/Dockerfile (hunk 1 above)
Comment (bajajneha27) [reply .kokoro/continuous/Dockerfile:1]: I'm not sure about that either 😅  , I took whatever was there in `google-cloud-ruby` . I can remove it if that makes sense.

Comment (dazuma) [reply .kokoro/continuous/Dockerfile:1]: There isn't a `.kokoro/continuous/Dockerfile` in google-cloud-ruby

Comment (bajajneha27) [reply .kokoro/continuous/Dockerfile:1]: Oh yeah, I just noticed that it was there in my local only. 🤷🏻

**Pull request merged**